Feature/intro

mjjang88 opened the pull request:
1. 앱 진입 스플래시 화면 구현
2. 임시 HomeFragment 추가 (자동 생성됨)

mjjang88 committed:
PR Test

mjjang88 committed:
1. 앱 진입 스플래시 화면 구현
2. 임시 HomeFragment 추가 (자동 생성됨)

SongSieun commented on app/build.gradle:
```diff
@@ -43,6 +43,7 @@ android {
 }
 
 dependencies {
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
얘는 어디서 쓰이는 걸까요?/

mjjang88 replied:
auto import? 하다가 들어간 것 같은데, 사용되지 않는 것 같습니다.
삭제 처리했어요~

SongSieun commented on app/src/main/java/com/depromeet/muyaho/LaunchSplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class LaunchSplashActivity : AppCompatActivity() {
```
그냥 Splash로 해도 될것같은데 LaunchSplash 로 하신 이유가 있을까요?

mjjang88 replied:
좀 자세히 적어볼까 해서 작성했던 거고
저도 평소에 Splash라고 작성해서 Splash로 변경해 놓을게요

SongSieun commented on app/src/main/java/com/depromeet/muyaho/LaunchSplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class LaunchSplashActivity : AppCompatActivity() {
```
BaseActivity, BaseFragment, BaseViewModel 만들어야할것 같은데 같이 논의 해봐야겠네염 ㅜㅜ..

SongSieun commented on app/src/main/java/com/depromeet/muyaho/LaunchSplashActivity.kt:
```diff
@@ -0,0 +1,27 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        lifecycleScope.launch(Dispatchers.Default) {
+            // 초기화 작업은 여기에서 진행
+            delay(2000)
```
초기화 작업은 viewmodel 로 옮기는게 좋을것 같아요~

mjjang88 replied:
viewmodel의 function 콜은 액티비티에서 처리하지 않나요?
저는 저기에서 viewmodel.xxx() 와 같은 형태로 사용하는 걸로 생각하고 있었어요.

SongSieun replied:
viewmodel에 init{} 사용하면돼요!

mjjang88 replied:
이해 했습니다.

SongSieun commented on app/src/main/java/com/depromeet/muyaho/MainActivity.kt:
```diff
@@ -1,4 +1,4 @@
-package com.depromeet.muyaho.activity
+package com.depromeet.muyaho
```
main? home? 탭별로 패키지 나누는게 어떨까용?

mjjang88 replied:
패키지 쪼개는 쪽으로 가는거 찬성입니다.

SongSieun replied:
home 패키지로 분리 부탁드릴게용!

mjjang88 replied:
저 MainActivity는 Single Activity 구조에 사용될 단 하나의 액티비티라서요.
메인 화면을 보여주는 대상은 아닙니당.

SongSieun replied:
아하 그렇겠네용

mjjang88 committed:
1.사용하지 않는 lib dependency 제거
2.액티비티명 변경 (LaunchSplashActivity -> SplashActivity)
3.activity hilt annotation 작성

mjjang88 committed:
HomeFragment 패키지명 변경

mjjang88 committed:
SplashViewModel 생성 / splash 패키지 생성
